lib/web/ui/git_server.go

```diff
@@ -78,3 +80,52 @@ func MakeGitServer(clusterName string, server types.Server, requiresRequest bool
 	}
 	return uiServer
 }
+
+// CreateGitServerRequest is a request to create a Git server.
+type CreateGitServerRequest struct {
+	// SubKind is a git server subkind such as GitHub
+	SubKind string `json:"subKind"`
+	// Name is the git server name
+	Name string `json:"id"`
+	// GitHub contains metadata for GitHub proxy severs.
+	GitHub *GitHubServerMetadata `json:"github,omitempty"`
+	// Overwrite performs an upsert of the resource.
+	Overwrite bool `json:"overwrite,omitempty"`
+}
+
+// Check checks if the provided values are valid.
+func (r *CreateGitServerRequest) Check() error {
```

Comment:
just wanted to double check the create result of git server in the UI:

```
kind: git_server
metadata:
  labels:
    teleport.internal/github-org: goteleport-core-test
  name: goteleport-core-test
  revision: 7ebb2279-5517-408c-9546-78fcc22f0722
spec:
  addr: github.com:22
  github:
    integration: goteleport-core-test
    organization: goteleport-core-test
  hostname: goteleport-core-test.teleport-github-org
  rotation:
    current_id: ""
    last_rotated: "0001-01-01T00:00:00Z"
    schedule:
      standby: "0001-01-01T00:00:00Z"
      update_clients: "0001-01-01T00:00:00Z"
      update_servers: "0001-01-01T00:00:00Z"
    started: "0001-01-01T00:00:00Z"
  version: ""
sub_kind: github
version: v2
```

Reply:
The name of the git_server resource does not show up anywhere in the UI (since not important). So i think we make the name the same as the integration name for easier search down the road.